src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -715,6 +715,25 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'draw.io', 'xwiki-events-bridge'
     oldImportFiles.apply(this, newArgs);
   }
 
+  //
+  // Change the URL generated by export for containing the actual host.
+  //
+  var oldCreateLink = EditorUi.prototype.createLink;
+  EditorUi.prototype.createLink = function() {
+    let rawURL = oldCreateLink.apply(this, arguments);
+    // Do not include '#' because it's automatically added by getURL function below.
+    let documentFragment = rawURL.substring(rawURL.indexOf('#') + 1);
+    if (documentFragment.substring(0, 1) == 'R') {
+      let queryString = rawURL.substring (rawURL.indexOf('/?'), rawURL.indexOf('#R'));
+      // Append source parameter to the query
+      // Do not append '?' for an empty queryString because it's automatically added by getURL function
+      queryString = ((queryString != '') ? queryString + '&amp;source=url' : queryString + 'source=url');
+      return  window.location.protocol + window.location.host +
```

Comment:
Does it work if you return a relative URL? (i.e. if you don't specify the protocol and host). It should normally.